recipes/zstd/all/conanfile.py

```diff
@@ -90,3 +102,8 @@ def package_info(self):
         self.cpp_info.components["zstdlib"].libs = collect_libs(self)
         if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.components["zstdlib"].system_libs.append("pthread")
+
+        if self.options.build_programs:
+            # TODO: Remove after dropping Conan 1.x from ConanCenterIndex
+            bindir = os.path.join(self.package_folder, "bin")
+            self.runenv_info.prepend_path("PATH", bindir)
```

Comment:
Does conan detects that `self.package_folder/bin` exists, and add the path to self.runenv_info automatically? 

@uilianries is this actually necessary?

Reply:
This is necessary for the test package to be able to run the produced executable. If We did not do this and the build system has zstd installed in the OS then the `self.run(..., env=conanrun)` in the test package will use OS installed one instead of the one produced here!

Reply:
@klausholstjacobsen It's not needed actually, the zstd can be added as VirtualBuildEnv, tool_requires, so it should be available. Real example: https://github.com/conan-io/conan-center-index/blob/230ab447b90a136dca3b2bf45875b39c62136103/recipes/yasm/all/test_package/conanfile.py#L6

Reply:
Gotcha!

Reply:
@klausholstjacobsen I'll verify locally

Reply:
I also have problem making VirtualBuildEnv and tool_requires work with conan 2.0.9:
It seem to calculate different required package id for the requires(...) and tool__requires(...)

See here:
```
zstd/1.5.5: Created package revision 1c38485aeb7744c655abe5f56c5f1c89
zstd/1.5.5: Package '43c10c705bb0f1d0030e597cb09a05ef5953fed9' created
zstd/1.5.5: Full package reference: zstd/1.5.5#4e5fbc9a14e898483676fb77fb4eef3b:43c10c705bb0f1d0030e597cb09a05ef5953fed9#1c38485aeb7744c655abe5f56c5f1c89
zstd/1.5.5: Package folder [hidden]/.conan2/p/b/zstd9d8fd2314131c/p
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: zstd/1.5.5
WARN: deprecated:     'env_info' used in: zstd/1.5.5

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    zstd/1.5.5 (test package): [hidden]/source/conan-center-index/recipes/zstd/all/test_package/conanfile.py
Requirements
    zstd/1.5.5#4e5fbc9a14e898483676fb77fb4eef3b - Cache
Build requirements
    zstd/1.5.5#4e5fbc9a14e898483676fb77fb4eef3b - Cache

======== Computing necessary packages ========
Requirements
    zstd/1.5.5#4e5fbc9a14e898483676fb77fb4eef3b:43c10c705bb0f1d0030e597cb09a05ef5953fed9#1c38485aeb7744c655abe5f56c5f1c89 - Cache
Build requirements
    zstd/1.5.5#4e5fbc9a14e898483676fb77fb4eef3b:c1b14c3945d5adf1e5ae440c43c12f88117f2edc - Missing

======== Installing packages ========
ERROR: Missing binary: zstd/1.5.5:c1b14c3945d5adf1e5ae440c43c12f88117f2edc

zstd/1.5.5: WARN: Can't find a 'zstd/1.5.5' package binary 'c1b14c3945d5adf1e5ae440c43c12f88117f2edc' for the configuration:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.version=11
os=Linux
[options]
build_programs=True
fPIC=True
shared=False
threading=True

ERROR: Missing prebuilt package for 'zstd/1.5.5'
Check the available packages using 'conan list zstd/1.5.5:* -r=remote'
'conan test' tested packages must exist, and '--build' argument is used only for the 'test_package' dependencies, not for the tested dependencies

More Info at 'https://docs.conan.io/2/knowledge/faq.html#error-missing-prebuilt-package'
```

build command:
`conan create recipes/zstd/all/ -pr:h default -pr:b default --version 1.5.5 --build zstd/* --build missing -o:h *:shared=True`

As you can see the normal requires in the test packages resolves package id 43xxx and the tool_requires resolves c1xxx

Reply:
Never got the above "multiple package id" issue solved.
But found a workaround from other inspirational packages.

In stead of doing this: 
```
    def requirements(self):
        self.requires(self.tested_reference_str)

    def build_requirements(self):
        self.tool_requires(self.tested_reference_str)
```
I am doing this:
```
    def requirements(self):
        self.requires(self.tested_reference_str, run=True)
```
I guess this is due to the fact that `package_type = "library"` even though the package also produces binaries. But I cannot figure out how that translates to conan reolving mulitple dependent package versions of the package.